src/pages/address/[principal].tsx

```diff
@@ -0,0 +1,168 @@
+import useSWR from 'swr';
+import {
+  AddressBalanceResponse,
+  MempoolTransaction,
+  MempoolTransactionListResponse,
+  TransactionResults,
+} from '@blockstack/stacks-blockchain-api-types';
+import { Box, Flex, Grid } from '@stacks/ui';
+import { Title } from '@components/typography';
+import type { NextPage, NextPageContext } from 'next';
+import { PageWrapper } from '@components/page';
+import { Section } from '@components/section';
+import { Rows } from '@components/rows';
+import { TransactionList } from '@components/transaction-list';
+import { microToStacks, truncateMiddle, validateStacksAddress } from '@common/utils';
+import { useApiServer } from '@common/hooks/use-api';
+import { getServerSideApiServer } from '@common/api/utils';
+import { fetchAllAccountData } from '@common/api/accounts';
+import { TokenBalancesRow } from '@components/balances/token-balances-row';
+import { Activity } from '@components/activity-row';
+import { StxBalances } from '@components/balances/stx-balance-card';
+import { getStackStartBlockHeight, hasTokenBalance } from '@common/utils/accounts';
+import { TokenBalancesCard } from '@components/balances/principal-token-balances';
+import { Meta } from '@components/meta-head';
+import { AddressNotFound } from '@components/address-not-found';
+
+const SummaryCard = ({ principal, hasTokenBalances, data }: any) => {
+  return (
+    <Section mb={'extra-loose'} title="Summary">
+      <Rows
+        px="base"
+        noTopBorder
+        items={[
+          {
+            label: {
+              children: 'Address',
+            },
+            children: principal,
+            copy: principal,
+          },
+          {
+            condition: !!data?.transactions?.results?.length,
+            label: {
+              children: 'Activity',
+            },
+            children: <Activity txs={data?.transactions?.results} />,
+          },
+          {
+            condition: !!hasTokenBalances,
+            label: {
+              children: 'Holdings',
+            },
+            children: <TokenBalancesRow balances={data?.balances} />,
+          },
+          {
+            label: {
+              children: 'Fees',
+            },
+            children: `${microToStacks(data?.balances?.stx?.total_fees_sent || 0)} STX`,
+          },
+        ]}
+      />
+    </Section>
+  );
+};
+
+interface AddressPageData {
+  principal: string;
+  balances?: AddressBalanceResponse;
+  transactions?: TransactionResults | MempoolTransactionListResponse;
+  pendingTransactions?: MempoolTransaction[];
+  error?: boolean;
+}
+
+const AddressPage: NextPage<AddressPageData> = props => {
+  const { principal, balances, pendingTransactions, transactions, error } = props;
+  if (error) {
+    return (
+      <>
+        <Meta title="Address not found" />
+        <AddressNotFound />
+      </>
+    );
+  }
+  const apiServer = useApiServer();
+  const { data } = useSWR(principal, fetchAllAccountData(apiServer as any), {
```

Comment:
The types here are pretty messy, which comes from the `initialData` setup - where the types don't match. Again, if we could tweak these TS types, it would be much nicer.

Basically, you should be able to change this to `useSWR<AllAccountData>` and have it just work, but it won't due to the `initialData` setup.